### ISSUE TITLE
Using test.py on Google Colab

### ISSUE THREAD

**Issue by axlecky** [eriklindernoren/PyTorch-YOLOv3#774]
Hi, I'm trying to use the old test.py for the test set with my trained Tiny YOLOv3 model using the code:`!python3 test.py --task test --weights weights/last.pt --data /content/yolov3/data/roboflow.data --cfg '/content/yolov3/cfg/yolov3-tiny.cfg'`

However. the test.py keeps running on my validation set and not the test set, is there any way to include a file to direct it to the test set instead? Thanks

**Comment by Flova** [comment]
The test.py reads the `*.data` file specified with `-d` and it contains the paths for the test, train and val dataset. By default the validation dataset. If you want to use a validation dataset instead you would need to change this line https://github.com/eriklindernoren/PyTorch-YOLOv3/blob/master/pytorchyolo/test.py#L181. We could add a check if the `.data` file has a test set and prioritize it over the validation.

**Comment by axlecky** [comment]
thanks! solved the issue by adding a separate .data file with the .txt file for the test images

**Comment by jaagut** [comment]
> We could add a check if the .data file has a test set and prioritize it over the validation. Or we add a comand line param to select test/val

I'd prefer the first solution

**Comment by Flova** [comment]
I would prefer the later one, as the test set should only be used for testing in the end and not the default for all r&d evaluations.
